packages/fortinet_fortimail/data_stream/log/elasticsearch/ingest_pipeline/pipeline_system.yml

```
@@ -0,0 +1,171 @@
+      allow_duplicates: false
+  - append:
+      field: event.category
+      value: configuration
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'config'
+      allow_duplicates: false
+  - set:
+      field: event.type
+      value: [change]
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'config'
+  - set:
+      field: event.type
+      value: [deletion]
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'config' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('delet')
+      override: true
+  - set:
+      field: event.category
+      value: [host]
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'system' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('system')
+  - set:
+      field: event.category
+      value: [configuration]
+      if: (ctx.fortinet_fortimail?.log?.sub_type == 'system' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('change')) || (ctx.fortinet_fortimail?.log?.sub_type == 'update' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('update'))
+      override: true
+  - set:
+      field: event.type
+      value: [info]
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'system'
+  - set:
+      field: event.type
+      value: [end]
+      if: ctx.fortinet_fortimail?.log?.sub_type == 'system' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('shutdown')
+      override: true
+  - set:
+      field: event.type
+      value: [change]
+      if: (ctx.fortinet_fortimail?.log?.sub_type == 'system' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('change')) || (ctx.fortinet_fortimail?.log?.sub_type == 'update' && ctx.fortinet_fortimail.log.message.toLowerCase().contains('update'))
+      override: true
+  - append:
+      field: fortinet_fortimail.log.user
+      value: '{{{fortinet_fortimail.log.log_details.user}}}'
+      allow_duplicates: false
+      if: ctx.fortinet_fortimail?.log?.log_details?.user != null
+  - append:
+      field: user.name
+      value: '{{{fortinet_fortimail.log.log_details.user}}}'
+      allow_duplicates: false
+      if: ctx.fortinet_fortimail?.log?.log_details?.user != null
+  - append:
+      field: related.user
+      value: '{{{fortinet_fortimail.log.log_details.user}}}'
+      allow_duplicates: false
+      if: ctx.fortinet_fortimail?.log?.log_details?.user != null
+  - grok:
+      field: message
+      ignore_missing: true
+      ignore_failure: true
+      patterns:
+        - '^%{DATA}[I|i]nterface %{NUMBER:fortinet_fortimail.log.port:long}%{DATA} [U|u]ser %{NOTSPACE:_tmp.user} %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$'
+        - '^%{DATA}[I|i]nterface %{NUMBER:fortinet_fortimail.log.port:long}%{DATA} [U|u]ser %{NOTSPACE:_tmp.user}%{GREEDYDATA:_tmp.msg}$'
+        - '^%{DATA}[U|u]ser %{NOTSPACE:_tmp.user} %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$'
+        - '^%{DATA}[U|u]ser %{NOTSPACE:_tmp.user} %{GREEDYDATA:_tmp.msg}$'
```

Similar reductions and corrections.
```suggestion
        - '^%{PREFIX} %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$'
        - '^%{PREFIX}%{GREEDYDATA:_tmp.msg}$'
        - '^%{DATA}%{USER} %{DATA}%{IP:fortinet_fortimail.log.ip}%{GREEDYDATA:_tmp.msg}$'
        - '^%{DATA}%{USER} %{GREEDYDATA:_tmp.msg}$'
      pattern_definitions:
        PREFIX: '%{DATA}[Ii]nterface %{NUMBER:fortinet_fortimail.log.port:long}%{DATA} %{USER}'
        USER: '[Uu]ser %{NOTSPACE:_tmp.user}'
```
(untested)